src/server.rs

```diff
@@ -43,16 +49,44 @@ impl Server {
         Executor::new(self.node.clone(), cpu_memory).await;
     }
 
-    async fn run_server(&self) -> Result<(), Error> {
-        let port = self.port;
-        let address: SocketAddr = format!("{}:{}", LOCALHOST, port)
-            .parse()
-            .map_err(Error::Parse)?;
-        let mut listener = TcpListener::bind(&address).map_err(Error::Bind)?;
+    async fn run_server_unix(&self) -> Result<(), Error> {
+        let mut unix_socket = {
+            static PATH: &'static str = "/tmp/sock";
+            UnixListener::bind(&PATH).map_err(Error::Bind)?
+        };
+
+        let node_ref = Arc::new(self.node.clone());
```

Comment:
There is a lot of code duplication with `run_server_tcp`, which presents an opportunity for refactoring, potentially by using the enum.